src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1939,6 +2105,9 @@ open_common(int (*real_open)(const char *pathname, int oflags, ...), const char
 
 	if (!hook_enabled)
 		goto org_func;
+	/* special cases. Needed to avoid dead lock inside daos_init(). */
+	if (strstr(pathname, "/etc/libfabric.conf"))
+		goto org_func;
```

Comment:
In previous test, libfabric.so from build directory was loaded. It was trying to open configuration file from DFS inside daos_init(). Accessing one file on DAOS triggered calling daos_init().

Reply:
This might be fixed for now then with the build changes, but in general this is a thing that could happen so detecting it programatically and returning would be good so that if/when it does the code does the correct thing.